File: src/main/java/com/hrms/hr_payroll_management_system/service/CompanyService.java
package com.hrms.hr_payroll_management_system.service;

import com.hrms.hr_payroll_management_system.dto.request.company.CreateCompanyRequest;
import com.hrms.hr_payroll_management_system.dto.request.company.UpdateCompanyRequest;
import com.hrms.hr_payroll_management_system.dto.response.company.CompanyResponse;

import java.util.List;

public interface CompanyService {

    CompanyResponse create(
            CreateCompanyRequest request
    );

    List<CompanyResponse> getAll();

    CompanyResponse getById(Long id);

    CompanyResponse update(
            Long id,
            UpdateCompanyRequest request
    );

    void delete(Long id);
}

File: src/services/company.service.js
import { companyApi } from '../api/company.api';

export const companyService = {
  async list() {
    const res = await companyApi.list();
    return res.data;
  },
  async getById(id) {
    const res = await companyApi.getById(id);
    return res.data;
  },
  async create(payload) {
    const res = await companyApi.create(payload);
    return res.data;
  },
  async update(id, payload) {
    const res = await companyApi.update(id, payload);
    return res.data;
  },
  async remove(id) {
    await companyApi.remove(id);
  },
  async uploadLogo(id, file) {
  const res = await companyApi.uploadLogo(id, file);
  return res.data;
},
};